src/include/api/feature_vector_array.h

```diff
@@ -68,6 +68,12 @@ class FeatureVectorArray {
     feature_size_ = datatype_to_size(feature_type_);
   }
 
+    FeatureVectorArray(
```

Comment:
We should add overloads like this to the other type-erased api classes, e.g., `FlatL2Index`, `IVFFlatIndex`.  This would remove the need to explicitly convert to string at the call site.  See comments below.

The basic idea is that everything should work *as if* the automatic conversion from `std::filesystem::path` to `std::string` actually worked.

Some other classes might need overloads then as well -- `tdbMatrix` for instance.

If adding those overloads bloats the code too much, then we can revisit this and maybe leave some that don't have the overloads.

Reply:
I completely agree, I will have to clean this PR